### PR TITLE
Retry 50x

### PR DESCRIPTION
**Pull request by unicolet**
Add support for retrying the github request when it fails with 502.

### CODE REVIEW

File: lib/github_changelog_generator/octo_fetcher.rb [hunk 1]
```diff
@@ -373,14 +376,17 @@ def print_empty_line
...
+      if env_var.nil?
+        Helper.log.warn NO_TOKEN_PROVIDED
+      elsif env_var.include? ","
+        token = env_var.split(',')[@token_idx]
+        if token
+          Helper.log.warn "[OLD] Using token[#{@token_idx}]=#{token[0..5]}***********************************"
+          @token_idx = @token_idx + 1
+        else
+          sys_abort("No more github tokens to try, exiting")
+        end
```
Comment (unicolet) [inline lib/github_changelog_generator/octo_fetcher.rb:389]: FYI the above section was added by me to experiment with using multi-token in the same run, I'd like to leave it here for now. We can clean it up later if this PR improves the changelog generator stability :-)

Comment (dawn-minion) [reply lib/github_changelog_generator/octo_fetcher.rb:389]: @unicolet Nice, I like that there's native multi-tokens in it now, can drop the `changelog || changelog` block in the Jenkins job.

Comment (unicolet) [reply lib/github_changelog_generator/octo_fetcher.rb:389]: @dawn-minion question is, will it work? :sweat_smile: Last I tried it was not quite the success I hoped, but maybe I should give it another try now

File: Rakefile [hunk 2]
```diff
@@ -17,4 +17,4 @@ task :update_ssl_ca_file do
 end
 
 task checks: %i[rubocop rspec]
-task default: %i[rubocop rspec]
+task default: %i[rspec]
```
Comment (dawn-minion) [inline Rakefile:20]: @unicolet What was the reason for dropping the linter/formatter here? I haven't used Ruby/rubocop, was just curious =)

**Review by dawn-minion — APPROVED** [review]
One line to add in a handler for 502s? That's pretty cool. Looking forward to seeing it in action, and the token option too if it works. Just one question about the test/linter

File: Rakefile (hunk 2 above)
Comment (unicolet) [reply Rakefile:20]: @dawn-minion apparently the sources are messed up so I left it alone

Comment (dawn-minion) [reply Rakefile:20]: Ah ok - no worries then.

**Comment by unicolet** [comment]
Btw I've hacked a changelog job to use this version:

http://jenkins.ad1.proemion.com/job/update_changelog_experiment_umberto/

we should give it a try before rebuilding the `psp-build` image. Wdyt? If ok I'll switch it over :-)

**Comment by unicolet** [comment]
worked fine on a trial run. There were no errors from GH, so we don't know how much better this version is, but at least it's not worse :-)

@gkranis I'm going to merge today, do you think you can give it a look?

**Pull request merged**